### PR TITLE
Add full test cases for services.js

### PR DESCRIPTION
**Pull request by oott123**

### CODE REVIEW

File: test/services.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            };
+        });
+
+        afterEach(function () {
+            services.unRegisterAll();
+            services.destroy();
+        });
+
+        it('should register singleton', function () {
+            services.register('/foo', null, singletonService);
```
Comment (harttle) [inline test/services.js]: 其实不应该通过另一个模块来测试当前模块

File: Makefile [hunk 2]
```diff
@@ -11,7 +11,7 @@ export NAME = $(shell node -p 'require("./package.json").name')
 export VERSION = $(shell node -p 'require("./package.json").version')
 export DESCRIPTION = $(shell node -p 'require("./package.json").description')
 
-.PHONY: test dist doc
+.PHONY: test dist doc test-watch test-reports test-reports-ci
```
Comment (harttle) [inline Makefile]: 😀

File: src/services.js [hunk 3]
```diff
@@ -98,6 +98,7 @@ define(function (require) {
         }
 
         // 由于目前还是 URL 索引页面，ignoreCache 为同样 URL 创建新的页面仍不可行
+        // function getOrCreate(url)
```
Comment (harttle) [inline src/services.js]: What's this, WIP?

Comment (oott123) [reply src/services.js]: overload

**Comment by oott123** [comment]
@harttle review pls

**Pull request merged**